### ISSUE TITLE
Decoder Input 

### ISSUE THREAD

**Issue by HasnainKhanNiazi** [zhouhaoyi/Informer2020#431]
Hey guys, I really enjoyed reading the paper, and thanks for pushing the source code. I am working on Informer for a multivariate problem where I am having 94 features and one output target. I have a question about the model inference/prediction. In the _process_one_batch method, we are passing the decoder input where firstly, it is being initiated by zeros and it is being concatenated with batch_y. I am removing if conditions as I am working with padding==0. I am training for MS but I am not passing the target variable in the input by changing the read_data method so that seq_x will be having only the features.

    # decoder input
    dec_inp = torch.zeros([batch_y.shape[0], self.args.pred_len, batch_y.shape[-1]]).float()
    dec_inp = torch.cat([batch_y[:,:self.args.label_len,:], dec_inp], dim=1).float().to(self.device)

My question is why are we concatenating batch_y values as in realtime, we will not be having batch_y values. I trained a model and the results look so promising with the above decoder input but if I don't use the batch_y concatenation part then the results aren't looking good.

**Comment by HasnainKhanNiazi** [comment]
I have been debugging the code to find out more insights and as I am using MS mode that's why I changed the input in such a way that the target shouldn't be in the input; 

     if self.features=='M' or self.features=='MS':
            cols_data = df_raw.columns[1:-1]
            df_data - df_raw[cols_data]

As the last column is the target column so I am not passing that, now in training, I am doing the same and I have checked the complete code and it looks like the actual targets are not being used in training also.

       if self.scale:
            train_data = df_data[border1s[0]:border2s[0]]
            self.scaler.fit(train_data.values)
            data = self.scaler.transform(df_data.values)
        else:
            data = df_data.values
            
        df_stamp = df_raw[['date']][border1:border2]
        df_stamp['date'] = pd.to_datetime(df_stamp.date)
        data_stamp = time_features(df_stamp, timeenc=self.timeenc, freq=self.freq)

        self.data_x = data[border1:border2]
        if self.inverse:
            self.data_y = df_data.values[border1:border2]
        else:
            self.data_y = data[border1:border2]
        self.data_stamp = data_stamp

As it can be seen from the above code df_data is being used further to extract the x, and y features and in training for targets, there is zero padding being used but the strange thing is that the results look really really good. I am wondering if is there anything I am missing in the code where the targets are also being passed.
